### PR TITLE
dist: tools: coccinelle: add documentation

### PR DESCRIPTION
**Pull request by kaspar030**
This PR adds a README.md explaining some aspects of the CI's Coccinelle integration.

### CODE REVIEW

File: dist/tools/coccinelle/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## Overview
+
+Coccinelle allows testing C code against semantic patches, automatically
+creating diffs on the way. See http://coccinelle.lip6.fr/ for more information.
+
+## Coccinelle in RIOT
+
+RIOT uses Coccinelle as part of the CI static-tests as follows:
+
+- as with cppcheck, every added or modified file will be checked
```
Comment (miri64) [inline dist/tools/coccinelle/README.md]: I would at least want a link to the `cppcheck` path here. People coming fresh to the project might not know what you mean here.

**Comment by emmanuelsearch** [comment]
Looks good. ACK once @miri64 comment is addressed.

File: dist/tools/coccinelle/README.md (hunk 1 above)
Comment (kaspar030) [inline dist/tools/coccinelle/README.md]: I just removed the cppcheck half sentence. Even a link wouldn't explain much, as cppcheck has no documentation on the matter at all.

**Comment by kaspar030** [comment]
- rebased, addressed comment, squashed

**Comment by emmanuelsearch** [comment]
No need for Murdock (changes only adds a readme).
Go.

**Pull request merged**